chasm/test_component_test.go

```diff
@@ -0,0 +1,307 @@
+					},
+				},
+			},
+			Data: &commonpb.DataBlob{
+				EncodingType: 1,
+				Data:         []byte{0x42, 0x13, 0x73, 0x75, 0x62, 0x2d, 0x63, 0x6f, 0x6d, 0x70, 0x6f, 0x6e, 0x65, 0x6e, 0x74, 0x31, 0x2d, 0x64, 0x61, 0x74, 0x61},
+			},
+		},
+		"SubComponent1/SubComponent11": &persistencespb.ChasmNode{
+			Metadata: &persistencespb.ChasmNodeMetadata{
+				InitialVersionedTransition: &persistencespb.VersionedTransition{
+					NamespaceFailoverVersion: 1,
+					TransitionCount:          1,
+				},
+				LastUpdateVersionedTransition: &persistencespb.VersionedTransition{
+					NamespaceFailoverVersion: 2,
+					TransitionCount:          2,
+				},
+				Attributes: &persistencespb.ChasmNodeMetadata_ComponentAttributes{
+					ComponentAttributes: &persistencespb.ChasmComponentAttributes{
+						Type:  "test_sub_component_11",
+						Tasks: []*persistencespb.ChasmComponentAttributes_Task(nil),
+					},
+				},
+			},
+			Data: &commonpb.DataBlob{
+				EncodingType: 1,
+				Data:         []byte{0x42, 0x23, 0x73, 0x75, 0x62, 0x2d, 0x63, 0x6f, 0x6d, 0x70, 0x6f, 0x6e, 0x65, 0x6e, 0x74, 0x31, 0x2d, 0x73, 0x75, 0x62, 0x2d, 0x63, 0x6f, 0x6d, 0x70, 0x6f, 0x6e, 0x65, 0x6e, 0x74, 0x31, 0x31, 0x2d, 0x64, 0x61, 0x74, 0x61},
+			},
+		},
+		"SubComponent1/SubData11": &persistencespb.ChasmNode{
+			Metadata: &persistencespb.ChasmNodeMetadata{
+				InitialVersionedTransition: &persistencespb.VersionedTransition{
+					NamespaceFailoverVersion: 1,
+					TransitionCount:          1,
+				},
+				LastUpdateVersionedTransition: &persistencespb.VersionedTransition{
+					NamespaceFailoverVersion: 2,
+					TransitionCount:          2,
+				},
+				Attributes: &persistencespb.ChasmNodeMetadata_DataAttributes{
+					DataAttributes: &persistencespb.ChasmDataAttributes{},
+				},
+			},
+			Data: &commonpb.DataBlob{
+				EncodingType: 1,
+				Data:         []byte{0x42, 0x19, 0x73, 0x75, 0x62, 0x2d, 0x63, 0x6f, 0x6d, 0x70, 0x6f, 0x6e, 0x65, 0x6e, 0x74, 0x31, 0x2d, 0x73, 0x75, 0x62, 0x2d, 0x64, 0x61, 0x74, 0x61, 0x31, 0x31},
+			},
+		},
+		"SubData1": &persistencespb.ChasmNode{
+			Metadata: &persistencespb.ChasmNodeMetadata{
+				InitialVersionedTransition: &persistencespb.VersionedTransition{
+					NamespaceFailoverVersion: 1,
+					TransitionCount:          1,
+				},
+				LastUpdateVersionedTransition: &persistencespb.VersionedTransition{
+					NamespaceFailoverVersion: 2,
+					TransitionCount:          2,
+				},
+				Attributes: &persistencespb.ChasmNodeMetadata_DataAttributes{
+					DataAttributes: &persistencespb.ChasmDataAttributes{},
+				},
+			},
+			Data: &commonpb.DataBlob{
+				EncodingType: 1,
+				Data:         []byte{0x42, 0x9, 0x73, 0x75, 0x62, 0x2d, 0x64, 0x61, 0x74, 0x61, 0x31},
+			},
+		},
+	}
+
+	return serializedNodes
+}
+
+// Helper functions to regenerate testComponentMap body.
+// Use: generateMapInit(serializedNodes, "serializedNodes")
+func generateMapInit(m interface{}, mapName string) {
```

Comment:
is this for visualize serialized chasm nodes map for a component?

Reply:
This is to generate function above.